app/models/SpecialMentionGuarantee.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import models.ParseError.{AdditionalInfoInvalidCharacters, AdditionalInfoTooLong, AmountStringInvalid, AmountWithoutCurrency, CurrencyCodeInvalid}
+
+import scala.xml.NodeSeq
+
+sealed trait SpecialMention
+
+final case class SpecialMentionOther(xml: NodeSeq) extends SpecialMention
+
+final case class SpecialMentionGuarantee(additionalInfo: String) extends SpecialMention {
+  def toDetails(guaranteeReference: String) : Either[ParseError, SpecialMentionGuaranteeDetails] = {
+    if(additionalInfo.length > 18) return Left(AdditionalInfoTooLong("AdditionalInfo is too long"))
+    if(additionalInfo.matches("'[a-zA-Z0-9_.]*'")) return Left(AdditionalInfoInvalidCharacters("invalid characters in additional info"))
```

Comment:
You could do a single if check here. A min and max length can be applied to the regex and the error message could be changed to "Additional information must contain only letters and numbers and must have a length of between 1 and 18 characters."<br><br>The error messages returned with Left in the PR could be made more human readable.

Reply:
You aren't wrong, but I wasn't sure merging it was worthwhile.

Reply:
I'm resolving this as @jhkldev has said it was ok.